contractsSol5/KyberNetwork.sol

```diff
@@ -729,15 +729,13 @@ contract KyberNetwork is Withdrawable2, Utils4, IKyberNetwork, ReentrancyGuard {
                                 uint[] memory splitValuesBps)
         internal pure returns (uint srcAmount)
     {
-        uint destAmountSoFar;
-
-        for (uint i = 0; i < rates.length; i++) {
-            uint destAmountSplit = i == (splitValuesBps.length - 1) ?
-                (destAmount - destAmountSoFar) : splitValuesBps[i] * destAmount / BPS;
-            destAmountSoFar += destAmountSplit;
-
-            srcAmount += calcSrcQty(destAmountSplit, srcDecimals, destDecimals, rates[i]);
+        uint averageRate = 0;
+        for(uint i = 0; i < rates.length; i++) {
+            averageRate += splitValuesBps[i] * rates[i] / BPS;
         }
+        require(averageRate > 0, "average rate is 0");
```

Comment:
we can also calculate average rate from 
ActualDestAmount and actual srcAmount

Reply:
Yeah, using just `calcRateFromQty` might be enough